rest_framework_simplejwt/backends.py

```diff
@@ -31,6 +33,11 @@ def encode(self, payload):
         """
         Returns an encoded token for the given payload dictionary.
         """
+        if self.audience is not None:
```

Comment:
Actually @ajhodges .  I noticed one other thing.  Can you open another PR to address this?  The problem is that I'd like the `TokenBackend.encode` method to effectively be a constant method.  In other words, it shouldn't be mutating the values that are passed to it as is now being done.  So let's add something like this at the top of that method:
```python
payload_with_aud_iss = payload.copy()
```
...and then add the audience and issuer to `payload_with_aud_iss` and pass it to the underlying jwt lib.